### PR TITLE
Switch to bwrap

### LINKED ISSUE #1762: switch to bwrap
bwrap seems a better fit for the job of wrapping text: https://github.com/imichael2e2/bwrap

### PR DESCRIPTION
**Pull request by TheBlackSheep3**
This Pull Request fixes/closes #1762.

It changes the following:
- removed dependency textwrap
- added dependency bwrap 

I followed the checklist:
- [x] I extended unittests
- [x] I ran `make check` without errors
- [x] I tested the overall application
- [x] I added an appropriate item to the changelog

### CODE REVIEW

**Comment by TheBlackSheep3** [comment]
@AmmarAbouZor i thought it was easier to just approach it again from scratch. Now the macro `bwrap::wrap!` is used.

**Comment by AmmarAbouZor** [comment]
It looks simple and elegant now :100:

File: src/components/commit_details/details.rs [hunk 1]
```diff
@@ -86,11 +86,20 @@ impl DetailsComponent {
 		message: &CommitMessage,
 		width: usize,
 	) -> WrappedCommitMessage<'_> {
-		let wrapped_title = textwrap::wrap(&message.subject, width);
+		let width = if width < 1 { 1 } else { width };
```
Comment (extrawurst) [inline src/components/commit_details/details.rs:89]: lets use `width.max(1)` here. weird that clippy does not suggest this 🤔

File: src/components/commit_details/details.rs [hunk 2]
```diff
@@ -86,11 +86,20 @@ impl DetailsComponent {
...
+		let wrapped_title = bwrap::wrap!(&message.subject, width)
+			.split('\n')
```
Comment (extrawurst) [inline src/components/commit_details/details.rs:91]: are we sure os independent that line breaks here will always be just `\n` and not any of the other potential encodings?

File: src/components/commit_details/details.rs (hunk 1 above)
Comment (TheBlackSheep3) [reply src/components/commit_details/details.rs]: Thank you for the hint. It really sounds like something that should be recognized by clippy.

File: src/components/commit_details/details.rs (hunk 2 above)
Comment (TheBlackSheep3) [reply src/components/commit_details/details.rs]: Valid point. I think `.lines()` should address this issue.

**Comment by extrawurst** [comment]
@TheBlackSheep3 thank you ❤️

**Pull request merged**